### ISSUE TITLE
ListenableFutureExplained does not explain

### ISSUE THREAD

**Issue by alexanderkjeldaas** [google/guava#2732]
The reasons for using ListenableFuture over Future in the wiki can be summed up as:
  - "concurrency is hard".. and you need to use this powerful abstraction because "other code uses it".

This leaves the reader in a "wtf?" state of mind.

This is the section that needs to be improved, IMO:

"""

Concurrency is a hard problem, but it is significantly simplified by working with powerful and simple abstractions.[...]

We strongly advise that you always use ListenableFuture instead of Future in all of your code, because:
- Most Futures methods require it.
- It's easier than changing to ListenableFuture later.
- Providers of utility methods won't need to provide Future and ListenableFuture variants of their methods.

"""

https://github.com/google/guava/wiki/ListenableFutureExplained

**Comment by mkoryak** [comment]
The docs have been updated, I noticed.

**Comment by jbduncan** [comment]
@mkordas Hmm, I think I'm missing something, but I don't believe there have been any changes made yet to the docs that improve the situation compared to when this issue was raised. Can you clarify for me what text in the docs you think suggests that this issue is solved?

**Comment by alexanderkjeldaas** [comment]
Maybe this sentence can fix the issue:

"A ListenableFuture allows you to register callbacks to be executed once the computation is complete, or if the computation is already complete, immediately."

I understand that the ability to run a computation immediately might potentially make a more powerful abstraction, but the precise guarantees for *immediately* needs to be spelled out, as well how `Future` is less *immediate*.  Are these callbacks guaranteed to be run in the Executor of the computation in some cases? Are the callbacks guaranteed to be executed in sequence so there cannot be races between them?  Does the original computation wait for the callbacks to finish?  What specifically is the difference?

**Comment by cgdecker** [comment]
Yeah, the section at the top of that page seems to explain why proactively making use of the type in your code even if you could currently get away with just using plain `Future` is good, but it doesn't first explain why the `ListenableFuture` type _itself_ is good, which choosing to use it is obviously predicated on.
